src/lightning/pytorch/core/saving.py

```diff
@@ -157,20 +157,26 @@ def _load_state(
 
     obj = cls(**_cls_kwargs)
 
+    if isinstance(obj, pl.LightningDataModule):
+        if obj.__class__.__qualname__ in checkpoint:
+            obj.load_state_dict(checkpoint[obj.__class__.__qualname__])
+        return obj
+
     if isinstance(obj, pl.LightningModule):
+        if obj._strict_loading is not None and strict is not None and strict != obj.strict_loading:
+            raise ValueError(
```

Comment:
Why is this an error instead of a warning? Couldn't it be valid to choose a different value in `load_from_checkpoint`?

Reply:
I think this is a reasonably important conflict to be resolved explicitly by the user. Note that both options are default None, and only when both are explicitly set to opposite bools is there a conflict. Even if we warn, we would have to make a choice which setting has precedence, and if we are "wrong" then there will be a loading error that shadows our warning.